src/formatDate.js

```diff
@@ -8,7 +8,41 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat[fromFormat.length - 1];
+  const separator2 = toFormat[toFormat.length - 1];
+
+  const curDate = date.split(separator);
+  const dateObj = {};
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateObj[fromFormat[i]] = curDate[i];
+  }
+
+  const result = [];
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    if (toFormat[i] === 'YYYY' && fromFormat.indexOf('YY') !== -1) {
+      let curYear = dateObj['YY'];
+
+      if (curYear < 30) {
+        curYear = '20' + curYear;
+      } else {
+        curYear = '19' + curYear;
+      }
+
+      result.push(curYear);
+      continue;
+    }
+
+    if (toFormat[i] === 'YY' && fromFormat.indexOf('YYYY') !== -1) {
+      result.push(dateObj['YYYY'].slice(-2));
+      continue;
+    }
+
+    result.push(dateObj[toFormat[i]]);
+  }
+
+  return result.join(separator2);
```

Comment:
The `formatDate` function implementation seems correct, but there are a few edge cases that are not covered. For example, the function does not check if the input date string actually matches the `fromFormat` pattern or if the date parts (like month or day) are within valid ranges. Additionally, the function assumes that the year will always be the first or the last part of the date, which might not be the case for different locales or formats. Consider adding validation for the input date string and handling different positions of the year in the format array.